Mathlib/RingTheory/Derivation/Basic.lean

```diff
@@ -336,6 +336,65 @@ end RestrictScalars
 
 end
 
+section Lift
+
+variable {R : Type*} {A : Type*} {M : Type*}
+variable [CommSemiring R] [CommRing A] [CommRing M]
+variable [Algebra R A] [Algebra R M]
+variable {F : Type*} [FunLike F A M] [AlgHomClass F R A M]
+
+/--
+Lift a derivation via an algebra homomorphism `f` with a right inverse. This gives the derivation
+`f ∘ d ∘ f⁻¹`.
+-/
+def liftOfRightInverse (f : F) (f_inv : M → A) (hf : Function.RightInverse f_inv f)
+    (d : Derivation R A A) (hd : ∀ x, f x = 0 → f (d x) = 0) : Derivation R M M where
+  toFun x := f (d (f_inv x))
+  map_add' x y := by
+    suffices f (d (f_inv (x + y) - (f_inv x + f_inv y))) = 0 by simpa [sub_eq_zero]
+    apply hd
+    simp [hf _]
+  map_smul' x y := by
+    suffices f (d (f_inv (x • y) - x • f_inv y)) = 0 by simpa [sub_eq_zero]
+    apply hd
+    simp [hf _]
+  map_one_eq_zero' := by
+    suffices f (d (f_inv 1 - 1)) = 0 by simpa [sub_eq_zero]
+    apply hd
+    simp [hf _]
+  leibniz' x y := by
+    suffices f (d (f_inv (x * y) - f_inv x * f_inv y)) = 0 by simpa [sub_eq_zero, hf _]
+    apply hd
+    simp [hf _]
+
+@[simp]
+lemma liftOfRightInverse_apply (f : F) (f_inv : M → A) (hf : Function.RightInverse f_inv f)
+    (d : Derivation R A A) (hd : ∀ x, f x = 0 → f (d x) = 0) (x : A) :
```

Comment:
```suggestion
lemma liftOfRightInverse_apply {f : F} {f_inv : M → A} (hf : Function.RightInverse f_inv f)
    {d : Derivation R A A} (hd : ∀ x, f x = 0 → f (d x) = 0) (x : A) :
```
etc etc. The idea is that you don't explicitly need to ask for inputs if they can be inferred from later inputs.